src/main/resources/org/jenkinsci/plugins/workflow/job/console/NewNodeConsoleNote/script.js

```diff
@@ -14,9 +14,18 @@ Behaviour.specify("span.pipeline-new-node", 'NewNodeConsoleNote', 0, function(e)
     var nodeId = e.getAttribute('nodeId')
     var startId = e.getAttribute('startId')
     if (startId == null || startId == nodeId) {
-        e.innerHTML = e.innerHTML.replace(/.+/, '$&<span class="pipeline-show-hide"> (<a href="#" onclick="showHidePipelineSection(this); return false">hide</a>)</span>')
+        e.innerHTML = e.innerHTML.replace(/.+/, '$&<span class="pipeline-show-hide"> (<a href="#" class="pipeline-toggle">hide</a>)</span>')
         // TODO automatically hide second and subsequent branches: namely, in case a node has the same parent as an earlier one
+
+        const toggle = e.querySelector('.pipeline-toggle');
```

Comment:
Are you sure this is always matching the `<span class="pipeline-show-hide">` that we just added on line 17 above? That code is appending `<span class="pipeline-show-hide">` elements as children of `e`, so I am wondering if there might ever be two such children or if there is always one as in your test. Did you try testing nested `parallel` blocks?

Reply:
@basil 

using the below script configuration 

```
parallel first: {
    stage('First Nested') {
        parallel nestedA: {}, nestedB: {}
    }
}, second: {
    stage('"full" details') {
        parallel nestedC: {}, nestedD: {}
    }
}
```

I recorded the behavior of the toggle links:

https://www.loom.com/share/2acb8c6b041a449793132e3f4a433d70


Everything seems to be working okay. Can you confirm if this is the intended behavior?

Reply:
OK, I am glad the testing is passing. To make this bulletproof, I think we should select the last child with the `pipeline-show-hide` class rather than the first. Since the `/.+/` regular expression is always appending the new `<span>` element to the end of the list of child nodes, selecting the last child should always match the node we just added.

Reply:
Thanks for explaining that detail. New commit addresses this